client/stats/sender.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package stats
+
+import (
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/openvpn"
+	"github.com/mysterium/node/openvpn/middlewares/client/bytescount"
+	"github.com/mysterium/node/server"
+	"github.com/mysterium/node/server/dto"
+	"github.com/mysterium/node/session"
+)
+
+const statsSenderLogPrefix = "[session-stats-sender] "
+
+// RemoteStatsSender sends session stats to remote API server with a fixed sendInterval.
+// Extra one send will be done on session disconnect.
+type RemoteStatsSender struct {
+	sessionID       session.SessionID
+	providerID      identity.Identity
+	consumerCountry string
+
+	client      server.Client
+	signer      identity.Signer
+	statsKeeper bytescount.SessionStatsKeeper
```

Comment:
Is it possible to move SessionStatsKeeper from middleware too? It would be nice improvement